packages/block-editor/src/hooks/use-cached-truthy.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Keeps an up-to-date copy of the passed value and returns it. If value becomes falsy, it will return the last truthy copy.
+ *
+ * @param {any} value
+ * @return {any} value
+ */
+export function useCachedTruthy( value ) {
```

Comment:
This hook seem a bit misplaced. If we need to be a public API, it should be on the "compose" package (more generic as it has nothing to do with block-editor). But personally, I see it only used once and I'm not sure this hook is very valuable outside that use-case so why not just move it close to its current usage?